intelmq/bots/parsers/malwaredomainlist/parser.py

```diff
@@ -1,50 +1,42 @@
 # -*- coding: utf-8 -*-
-import csv
-import io
 import sys
 from urllib.parse import urlparse
 
-from intelmq.lib import utils
-from intelmq.lib.bot import Bot
+from intelmq.lib.bot import ParserBot
 from intelmq.lib.exceptions import InvalidValue
 from intelmq.lib.message import Event
 
 
-class MalwareDomainListParserBot(Bot):
+class MalwareDomainListParserBot(ParserBot):
     def add_http(self, url):
         """
         We can assume here, that everything is http by default if not given.
         """
         if '://' not in url:
             return 'http://' + url
 
-    def process(self):
-        report = self.receive_message()
-
-        raw_report = utils.base64_decode(report.get("raw"))
-        for row in csv.reader(io.StringIO(raw_report)):
-            event = Event(report)
-            self.logger.debug(repr(row))
-
-            event.add("time.source", row[0].replace('_', ' ') + " UTC")
-            if row[1] != '-':
-                event.add("source.url", self.add_http(row[1]))
-            try:
-                event.add("source.ip", row[2])
-            except InvalidValue:
-                event.add("source.url", self.add_http(row[2]))
-                event.add('source.ip', urlparse(row[2]).netloc)
-            event.add("source.reverse_dns", row[3])
-            event.add("event_description.text", row[4])
-            # TODO: ignore abuse contact for now
-            event.add("source.asn", int(row[6]))
-
-            event.add('classification.type', 'malware')
-            event.add("raw", ",".join(row))
-
-            self.send_message(event)
-        self.acknowledge_message()
-
+    parse = ParserBot.parse_csv
+    recoverline = ParserBot.recoverline_csv
+
+    def parseline(self, row, report):
+        event = Event(report)
+
+        event.add("time.source", row[0].replace('_', ' ') + " UTC")
+        if row[1] != '-':
+            event.add("source.url", self.add_http(row[1]))
+        try:
+            event.add("source.ip", row[2])
+        except InvalidValue:
+            event.add("source.url", self.add_http(row[2]))
+            event.add('source.ip', urlparse(row[2]).netloc)
+        event.add("source.reverse_dns", row[3])
+        event.add("event_description.text", row[4])
+        # TODO: ignore abuse contact for now
+        event.add("source.asn", int(row[6]))
+
+        event.add('classification.type', 'malware')
+        event.add("raw", ",".join(row))
```

Comment:
Could be done by recoverline (especially useful for non-trivial formats, as csv)